elevate-project/controllers/project.js

```diff
@@ -1,29 +1,89 @@
+ */
+
 const routeConfigs = require('../constants/routes')
 const requesters = require('../utils/requester')
-const requestParser = require('../utils/requestParser')
-
-// const createUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body)
-// }
-// const updateUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-
-// 	//const filteredRequestBody = requestParser.transformUpdateUserBody(req.body)
-// 	return await requesters.patch(req.baseUrl, selectedConfig.targetRoute.path, req.body, req.headers)
-// }
-
-// const entityTypeRead = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body, {
-// 		'X-auth-token': req.headers['x-auth-token'],
-// 	})
-// }
-
-const userController = {
-	// createUser,
-	// updateUser,
-	// entityTypeRead,
+const common = require('../constants/common')
+/**
+ * Fetch project templates from projects service.
+ * @name fetchProjectTemplates
+ * @param {Object} req
+ * @param {Object} res
+ * @param {Object} responses  
+ * @returns {JSON} - List of project templates
+ */
+/**
+ * req.body = {
+ * 		organization_id : 1,
+ * 		resourceType : ['survey','projects']
+ * }
+ */
+
+const fetchProjectTemplates = async (req, res, responses) => {
+	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
+
+	let response = { result: { data: [], count: 0 } }
+	let proceedToCallProjectService = false
+	let resp = {}
+
+	// fetch the max limit from the env file for the DB Find API
+	const max_limit = process.env.RESOURCE_MAX_FETCH_LIMIT ? parseInt(process.env.RESOURCE_MAX_FETCH_LIMIT, 10) : 1000
+
+	if (req.body) {
+		const resourceType = req?.body?.resourceType || [];
+		if (Array.isArray(resourceType) || resourceType.length > 0) {
+			const hasProjectType = resourceType.includes(common.RESOURCE_TYPE_PTOJECT);
+			proceedToCallProjectService = hasProjectType || resourceType.length === 0;
+		}
+	}
+
+	if (proceedToCallProjectService && req.headers['x-auth-token']) {
+		let body = {
+			"query": {
+				"status": common.PROJECT_STATUS_PUBLISHED
+			},
+			"projection": common.PROJECT_PROJECTION_FIELDS,
+			"limit": max_limit
+		}
+		const x_auth_token = req.headers['x-auth-token'].startsWith("Bearer ") || req.headers['x-auth-token'].startsWith("bearer ") ? req.headers['x-auth-token'].split(' ')[1] : req.headers['x-auth-token']
+		// custom header 
+		const header = {
+			'internal-access-token' : req.headers['internal_access_token'],
+			'X-auth-token': x_auth_token,
+			'Content-Type': 'application/json'
+		}
+		if (req.body.search) body.query.title = {
```

Comment:
@adithyadinesh0412  add like below to better readability
if (req.body?.search) {
  body.query.title = { $regex: req.body.search, $options: 'i' };
}